src/helpers/freeCompute.ts

```diff
@@ -1,35 +1,201 @@
+    }
+    throw e
+  }
+}
+
+export async function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+export async function checkComputeStatus(
+  nodeUrl: string,
+  jobId: string
+): Promise<ComputeStatus> {
+  const response = await axios.post(`${nodeUrl}/directCommand`, {
+    command: 'getComputeStatus',
+    jobId
+  })
+  return response.data[0]
+}
+
+export async function getComputeResult(
+  nodeUrl: string,
+  jobId: string,
+  consumerAddress: string,
+  signature: string,
+  index: number = 0,
+  nonce: number = 0
+) {
+  try {
+    console.log('Getting compute result for jobId:', jobId)
+    console.log('Using consumerAddress:', consumerAddress)
+    console.log('Using signature:', signature)
+    console.log('Using index:', index)
+    console.log('Using nonce:', nonce)
+
+    const response = await axios.post(`${nodeUrl}/directCommand`, {
+      command: 'getComputeResult',
+      jobId,
+      consumerAddress,
+      signature,
+      index,
+      nonce
+    })
+    return response.data
+  } catch (error) {
+    console.error('Error getting compute result:', error)
+    throw error
+  }
+}
 
+export async function saveResults(
+  results: any,
+  destinationFolder?: string
+): Promise<string> {
   try {
-    const response = await fetch(nodeUrl + '/directCommand', {
+    // Use provided destination folder or default to './results'
+    const resultsDir = destinationFolder || path.join(process.cwd(), 'results')
+
+    // Ensure results directory exists
+    if (!fs.existsSync(resultsDir)) {
+      console.log('Creating results directory at:', resultsDir)
+      await fs.promises.mkdir(resultsDir, { recursive: true })
+    }
+
+    const timestamp = new Date().toISOString().replace(/[:.]/g, '-')
+    const fileName = `compute-results-${timestamp}.txt`
+    const filePath = path.join(resultsDir, fileName)
+
+    console.log('Saving results to:', filePath)
+    console.log('Results content:', JSON.stringify(results, null, 2))
+
+    // Write the file
+    await fs.promises.writeFile(filePath, JSON.stringify(results, null, 2), 'utf-8')
+
+    // Verify file was created
+    if (!fs.existsSync(filePath)) {
+      throw new Error(`Failed to create file at ${filePath}`)
+    }
+
+    return filePath
+  } catch (error) {
+    console.error('Error saving results:', error)
+    console.error('Results directory:', destinationFolder || './results')
+    console.error('Results:', results)
+    throw new Error(`Failed to save results: ${error.message}`)
+  }
+}
+
+// Create output channel for compute logs
+export const computeLogsChannel = vscode.window.createOutputChannel('Ocean Compute Logs')
+
+export async function getComputeLogs(
+  nodeUrl: string,
+  jobId: string,
+  consumerAddress: string,
+  nonce: number,
+  signature: string
+): Promise<void> {
+  try {
+    // Make request to get compute logs
+    const response = await fetch(`${nodeUrl}/directCommand`, {
```

Comment:
There is not implemented in ocean.js right? Until then we will use fetch requests.

Reply:
Yeah, Alex said that ocean.js isn't ready yet so that's currently blocked. I'll do it on a separate PR